src/gui/userstatus.cpp

```diff
@@ -30,16 +30,37 @@ Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
 UserStatus::UserStatus(QObject *parent)
     : QObject(parent)
-    , _message("")
 {
 }
 
 UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
 {
+    // it needs to match the Status enum
+    const QHash<QString, Status> preDefinedStatus{{"online", Online},
+                                               {"dnd", DoNotDisturb}, //DoNotDisturb
+                                               {"away", Away},
+                                               {"offline", Offline},
+                                               {"invisible", Invisible}};
+    
     // api should return invisible, dnd,... toLower() it is to make sure 
     // it matches _preDefinedStatus, otherwise the default is online (0)
-    const auto statusEnum = _preDefinedStatus.value(status.isEmpty()? "online" : status.toLower(), 0);
-    return static_cast<Status>(statusEnum);
+    const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
+    return preDefinedStatus.value(statusKey, Online);
+}
+
+QString UserStatus::enumToUserString(Status status) const 
```

Comment:
Could also made into a static function of the cpp file. This way it would not "pollute" the class definition. A other question that I have is: Is the name of the function not to generic? For example I would choose `statusEnumToUserString()`.

Reply:
static functions are so like C++98 ;-)

But the general comment applies, it'd be better as a free function in an anonymous namespace at the beginning of the file.